pyccel/ast/core.py

```diff
@@ -5177,15 +5177,56 @@ def bodies(self):
         return b
 
 
-class IfTernaryOperator(If):
+class IfTernaryOperator(Basic, PyccelAstNode):
+    """Represents a IfTernaryOperator statement in the code.
 
-    """class for the Ternery operator"""
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : [condition , value_if_true, value_if_false]
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign, IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    """
     def __init__(self, *args):
-        for arg in self.args:
-            if len(arg[1].body)!=1:
-                raise TypeError('IfTernary body must be of length 1')
+        self._args = []
+        cond = args[0]
+        first = args[1]
+        second = args[2]
+        self._args = [cond, first, second]
```

Comment:
Why did you decide to use a list containing the three arguments rather than saving each into their own variable?

Reply:
Actually, I started using lists at first because I was not sure about what I should do. And I finished by keeping it Normally it should be  3 arguments.

Reply:
In that case I think it would be more readable if we use `self._cond` `self._first` `self._second`

Reply:
Or maybe even `self._cond`, `self._value_true`, `self._value_false`